File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.go
func maxProfit(prices []int) int {

	left := 0
	max := 0
	for r:=1; r < len(prices); r++{
		diference := prices[r] - prices[left]
		if diference > max{
			max = diference
		}
		if diference < 0{
			left = r
		}
		
	}
	return max
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        
        def recurse(node):
            if not node:
                return 0

            left = recurse(node.left)
            if left == -1:
                return -1

            right = recurse(node.right)
            if right == -1:
                return -1


            if abs(left - right) > 1:
                return -1

            return 1 + max(left, right)

        return recurse(root) != -1